Ieva stonkaite/feature/t 9 feature branch for pom refactoring

IevaStonkaite opened the pull request:
Adjusted the branch so that the the pull request points to my local develop at ieva_stonkaite/develop

Commit:
T1 and half of T2 refactoring

Commit:
Tasks T1 and T2 refactored

Commit:
T1 finished, T2 started in separate files

Commit:
Task1 (login.js) and task 2 (t2.js) refactored

Commit:
Deleted unnecessary file

DavidV44 commented on cypress/objects/t2Page.js:
```diff
@@ -0,0 +1,66 @@
+    }
+
+    
+
```
a bit too much of empty lines :)

DavidV44 commented on cypress/objects/t2Page.js:
```diff
@@ -0,0 +1,66 @@
+class T2Page {
```
This file should be name `timeLoggingPage `or something like that, because T2 doesn't say much :)

DavidV44 commented on cypress/objects/t2Page.js:
```diff
@@ -0,0 +1,66 @@
+        cy.visit('/'); 
+    }
+
+    getUserDropDown() {
```
This and some of other below methods are are in `loginPage.js` file. This file should only contain methods that work with elements of time logging page

DavidV44 commented on cypress/integration/t2.js:
```diff
@@ -0,0 +1,73 @@
+import T2Page from '../objects/t2Page';
+
+const t2Page = new T2Page();
```
you should name this `timeLoggingPage`, T2 is a bad name. Also you should import Login page and use those methods for login

Commit:
Created a file called "timeLoggingPage" and "loginPage" as advised;
Deleted repeated methods;
Deleted empty rows.

DavidV44 commented on cypress/integration/t2.js:
```diff
@@ -0,0 +1,68 @@
+
+const loginPage = new LoginPage();
+
+it('Should verify all user roles can log in and should see appropriate tabs - USER', function () {
```
I think these test should be in `login.js` file as they check login functionality

DavidV44 commented on cypress/objects/loginPage.js:
```diff
@@ -0,0 +1,42 @@
+        return cy.get(`[aria-label="${role}"]`)
+    }
+    
+    getUserTitle() {
```
Methods like these could be put in `common.js` or` pageComponents.js `files, because this is shared between pages and if you would want to check username or navigation bar it would look weird if you have to use `loginPage `to do so.

nikolaj-tolkaciov replied:
This is not yet fixed

DavidV44 commented on cypress/objects/timeLoggingPage.js:
```diff
@@ -0,0 +1,24 @@
+        return cy.get('.page__title')
+    }
+    
+    getUserTitle() {
```
This also should be put in some kind of shared file like `common.js`, see above comment

Commit:
Created a new file "common.js" with functions to get page title and user title;
Moved task1 and task2 into one file (login.js), instead two files, because they check were checking login functionality.

Commit:
Added more methods to common.js, deleted timeLoggingPage.js

Commit:
Additional fix for the urls

Commit:
Merge branch 'ieva_stonkaite/develop' into ieva_stonkaite/feature/T-9_feature_branch_for_POM_refactoring

# Conflicts:
#	cypress/integration/login.js

nikolaj-tolkaciov commented on cypress/integration/login.js:
```diff
@@ -1,103 +1,110 @@
-        cy.get('.tile.form').should('be.visible')
-        cy.get('.user-info__title').contains('Ieva Stonkaite');
-        cy.get('.main-nav').find('li').should('have.length', 2);
+        cy.wait(2000); // for reason js sometimes doesn't load my client
```
what is this about? it seems it work for others, try switching places with  common.getPageTitle().contains('Timesheets');

then it would wait for page title to change and afterwards assert URL, should be more stable that way, as URL changes during page loading and this might throw off cypress assert

nikolaj-tolkaciov commented on cypress/integration/login.js:
```diff
@@ -1,103 +1,110 @@
-    it('Should show calendar date', function () {
-        cy.get('.calendar--selected').find('span').contains(new Date().getDate());
+    it('Should validate what date is selected as “Today” on Time Logging page', function () {
+        common.getCalendarTodaysDate();
```
this function name is kind of misleading, because inside it actually assert as well, not only getting you todays date

nikolaj-tolkaciov commented on cypress/objects/common.js:
```diff
@@ -0,0 +1,40 @@
+    
+    getSpecificUrl(url) {
+        cy.visit(url); 
+    }
```
there is no point of having this function, as it just takes url and pass it to another function, you can call cy.visit(url) with same results

nikolaj-tolkaciov reviewed (changes requested):
Found few remarks, but other then that it is almost finished!
